test/integration/publishing/send-updates-calculation.test.js

```diff
@@ -0,0 +1,384 @@
+      const numberOfRecordsFunding = publishingConfig.dataPublishingMaxBatchSizePerDataSource
+      await db.calculation.create(mockCalculation1)
+      await db.funding.bulkCreate([...Array(numberOfRecordsFunding).keys()].map(x => { return { ...mockFunding1, fundingId: mockFunding1.fundingId + x, calculationId: mockCalculation1.calculationId } }))
+
+      await publish()
+
+      expect(mockSendMessage.mock.calls[0][0].body.fundings).toHaveLength(numberOfRecordsFunding)
+    })
+
+    test('should process calculation record when there are more funding records than publishingConfig.dataPublishingMaxBatchSizePerDataSource', async () => {
+      const numberOfRecordsCalculation = 1
+      const numberOfRecordsFunding = 1 + publishingConfig.dataPublishingMaxBatchSizePerDataSource
+      await db.calculation.create(mockCalculation1)
+      await db.funding.bulkCreate([...Array(numberOfRecordsFunding).keys()].map(x => { return { ...mockFunding1, fundingId: mockFunding1.fundingId + x, calculationId: mockCalculation1.calculationId } }))
+      const unpublishedBefore = await db.calculation.findAll({ where: { published: null } })
+
+      await publish()
+
+      const unpublishedAfter = await db.calculation.findAll({ where: { published: null } })
+      expect(unpublishedBefore).toHaveLength(numberOfRecordsCalculation)
+      expect(unpublishedAfter).toHaveLength(0)
+    })
+
+    test('should publish calculation record when there are more funding records than publishingConfig.dataPublishingMaxBatchSizePerDataSource', async () => {
+      const numberOfRecordsCalculation = 1
+      const numberOfRecordsFunding = 1 + publishingConfig.dataPublishingMaxBatchSizePerDataSource
+      await db.calculation.create(mockCalculation1)
+      await db.funding.bulkCreate([...Array(numberOfRecordsFunding).keys()].map(x => { return { ...mockFunding1, fundingId: mockFunding1.fundingId + x, calculationId: mockCalculation1.calculationId } }))
+
+      await publish()
+
+      expect(mockSendMessage).toHaveBeenCalledTimes(numberOfRecordsCalculation)
+      expect(mockSendMessage).not.toHaveBeenCalledTimes(numberOfRecordsFunding)
+    })
+
+    test('should publish all funding records when there are more funding records than publishingConfig.dataPublishingMaxBatchSizePerDataSource', async () => {
+      const numberOfRecordsFunding = 1 + publishingConfig.dataPublishingMaxBatchSizePerDataSource
+      await db.calculation.create(mockCalculation1)
+      await db.funding.bulkCreate([...Array(numberOfRecordsFunding).keys()].map(x => { return { ...mockFunding1, fundingId: mockFunding1.fundingId + x, calculationId: mockCalculation1.calculationId } }))
+
+      await publish()
+
+      expect(mockSendMessage.mock.calls[0][0].body.fundings).toHaveLength(numberOfRecordsFunding)
+      expect(mockSendMessage.mock.calls[0][0].body.fundings).not.toHaveLength(publishingConfig.dataPublishingMaxBatchSizePerDataSource)
+    })
+  })
+})
```

Comment:
Can we add test for 2 or more concurrent db.calculation.findAll() processes ?

Reply:
How do you propose that can be tested?